src/ActiveLogin.Authentication.BankId.AspNetCore/Areas/BankIdAuthentication/Views/BankId/_LoginScript.cshtml

```diff
@@ -69,27 +69,31 @@
             checkStatus(requestVerificationTokenElement.value, returnUrlElement.value, loginOptionsElement.value, orderRef);
         }
 
-        function login() {
+        function login(resetUi) {
             var requestVerificationTokenElement = formElement.querySelector('[name="RequestVerificationToken"]');
             var returnUrlElement = formElement.querySelector('[name="ReturnUrl"]');
             var cancelReturnUrlElement = formElement.querySelector('[name="CancelReturnUrl"]');
             var loginOptionsElement = formElement.querySelector('[name="LoginOptions"]');
             var personalIdentityNumberElement = formElement.querySelector('[name="PersonalIdentityNumber"]');
             var personalIdentityNumber = !!personalIdentityNumberElement ? personalIdentityNumberElement.value : '';
 
-            initialize(requestVerificationTokenElement.value, returnUrlElement.value, cancelReturnUrlElement.value, loginOptionsElement.value, personalIdentityNumber, personalIdentityNumberElement);
+            initialize(resetUi, requestVerificationTokenElement.value, returnUrlElement.value, cancelReturnUrlElement.value, loginOptionsElement.value, personalIdentityNumber, personalIdentityNumberElement);
         }
 
         // BankID
 
         var autoStartAttempts = 0;
         var loginIsCancelledByUser = false;
-        function initialize(requestVerificationToken, returnUrl, cancelUrl, loginOptions, personalIdentityNumber, personalIdentityNumberElement) {
+        function initialize(resetUi, requestVerificationToken, returnUrl, cancelUrl, loginOptions, personalIdentityNumber, personalIdentityNumberElement) {
             loginIsCancelledByUser = false;
-            formFieldsetElement.disabled = true;
-            show(formSubmitButtonSpinnerElement);
-            hide(formSubmitButtonTextElement);
-            hide(qrCodeElement);
+
+            if (resetUi) {
```

Comment:
Perhaps this is better off as a separate function that can be called before `login()` to avoid passing the flag? I guess such a function could be useful for cancelling and error handling as well.

Reply:
I like, will fix!